src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -7,292 +7,186 @@ import com.esotericsoftware.kryo.io.{Input, Output}
 import com.esotericsoftware.kryo.{Kryo, KryoSerializable}
 import is.hail.expr.types._
 import is.hail.utils._
+import is.hail.nativecode._
 import sun.reflect.generics.reflectiveObjects.NotImplementedException
 
 object Region {
   def apply(sizeHint: Long = 128): Region = {
-    new Region(new Array[Byte](sizeHint.toInt))
+    new Region()
   }
 
   def scoped[T](f: Region => T): T =
     using(Region())(f)
 }
 
-final class Region(
-  private var mem: Array[Byte],
-  private var end: Long = 0
-) extends UnKryoSerializable with AutoCloseable {
-  def size: Long = end
-
-  def capacity: Long = mem.length
-
-  def copyFrom(other: Region, readStart: Long, writeStart: Long, n: Long) {
-    assert(size <= capacity)
-    assert(other.size <= other.capacity)
-    assert(n >= 0)
-    assert(readStart >= 0 && readStart + n <= other.size)
-    assert(writeStart >= 0 && writeStart + n <= size)
-    Memory.memcpy(mem, writeStart, other.mem, readStart, n)
-  }
-
-  def loadInt(off: Long): Int = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 4 <= size)
-    Memory.loadInt(mem, off)
-  }
-
-  def loadLong(off: Long): Long = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 8 <= size)
-    Memory.loadLong(mem, off)
-  }
-
-  def loadFloat(off: Long): Float = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 4 <= size)
-    Memory.loadFloat(mem, off)
-  }
-
-  def loadDouble(off: Long): Double = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 8 <= size)
-    Memory.loadDouble(mem, off)
-  }
-
-  def loadAddress(off: Long): Long = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 8 <= size)
-    Memory.loadAddress(mem, off)
-  }
-
-  def loadByte(off: Long): Byte = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + 1 <= size)
-    Memory.loadByte(mem, off)
-  }
-
-  def loadBytes(off: Long, n: Int): Array[Byte] = {
-    assert(size <= capacity)
-    assert(off >= 0 && off + n <= size)
+// Off-heap implementation of Region refquires some interface changes
```

Comment:
Again, I don't like comments like this in the code.  After the changes are in place, this comment won't make sense without the diff context.

Reply:
ok